dealroom_cloud_run_auth/session.py

```diff
@@ -0,0 +1,226 @@
+import os
+from typing import Any, Callable
+from urllib.parse import urljoin
+
+from google.auth.transport import Response
+from google.auth.transport.requests import AuthorizedSession as DefaultAuthedSession
+
+from dealroom_cloud_run_auth.id_token_credentials import (
+    get_default_id_token_credentials,
+)
+
+URLJoiner = Callable[[str, str], str]
+Timeout = float | tuple[float, float]
+
+_DEFAULT_TIMEOUT = 120.0
+
+
+def default_url_joiner(base_url: str, path: str) -> str:
+    """Use `urllib.parse.urljoin` to join the base URL and the path."""
+    return urljoin(base_url + "/", path)
+
+
+def default_user_agent() -> str | None:
+    k_service = os.getenv("K_SERVICE")
+    k_revision = os.getenv("K_REVISION")
+    if k_service and k_revision:
+        return f"{k_service}/{k_revision}"
```

Comment:
wouldn't "/" be a problematic char for user-agent naming?

Reply:
No :) discovered that's the usual syntax for User-Agents https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#syntax